ast.ts

```diff
@@ -8,6 +8,8 @@ export type Type =
   | {tag: "class", name: string}
   | {tag: "either", left: Type, right: Type }
 
+export type SourceLocation = { line: number }
```

Comment:
I'm assuming this will get richer, but this is somewhat of a placeholder for now to get this in quickly?

Reply:
Yes, professor. This is just a placeholder. We will add more information as we go on (Eg: "column" as suggested in our previous pull request)